parse_date.re

```diff
@@ -707,24 +707,25 @@ static const timelib_tz_lookup_table* abbr_search(const char *word, timelib_long
 	return NULL;
 }
 
-static timelib_long timelib_lookup_abbr(char **ptr, int *dst, char **tz_abbr, int *found)
+#define TIMELIB_TP_VALUE(tp) tp->gmtoffset - (tp->type * 3600)
+
+static timelib_long timelib_lookup_abbr(char **ptr, int *dst, int is_abbrev, char **tz_abbr, int *found)
 {
 	char *word;
 	char *begin = *ptr, *end;
 	timelib_long  value = 0;
 	const timelib_tz_lookup_table *tp;
-
-	while (**ptr != '\0' && **ptr != ')' && **ptr != ' ') {
+	while (**ptr != '\0' && **ptr != ')' && **ptr != ' ' && ((is_abbrev) ? **ptr != '-' : true)) {
```

Comment:
There is no need to use a ternary here. What `is_abbrev` really does here, is to also allow `-` to be parsed in the string?